### PR TITLE
Copy code and create gradle build

### PR DESCRIPTION
**Pull request by dbmeneses**
I added sonar-enterprise as a remote and copied all the relevant commits (amending commits to only include files relevant to the new algorithm). I combined a few commits from the same author and I removed sonar ticket number from the messages.
I added a few commits to fix a few things and to sets up a basic gradle build.

Tests are passing but we need to setup CI/CD

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,25 @@
+plugins {
+    // Apply the java-library plugin for API and implementation separation.
+    id 'java-library'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(17)
```
Comment (leo-geoffroy-sonarsource) [inline build.gradle:8]: I think this should be 11, since scanner is not on 17

**Review by lukasz-jarocki-sonarsource — APPROVED** [review]
Building nicely on windows :) One test is not passing on Windows but it is fixed on the common branch.

File: build.gradle (hunk 1 above)
Comment (dbmeneses) [reply build.gradle:8]: good catch, I'll fix it

**Comment by lukasz-jarocki-sonarsource** [comment]
I pushed the unit tests to this branch too

**Pull request merged**